modules/dkimPolicy.jsm

```diff
@@ -409,6 +409,162 @@ var Policy = {
 		}
 	},
 
+	/**
+	 * Check that the list of signed headers satisfy our policies.
+	 * - Warn if recommended headers are not signed.
+	 * - Try detecting maliciously added unsigned headers.
+	 *
+	 * The detection for maliciously added unsigned headers 
+	 * only considers all the recommended signed headers.
+	 * It simply ensures that a message does not contain both 
+	 * signed and unsigned values of a header.
+	 * Using the same mechanism for all signed headers 
+	 * will cause problems there added headers is normal
+	 * e.g. the Received header.
+	 *
+ 	 * @param {MAP<String,String>} msgHeaders
+	 *			All headers of the message to check as headername-value-pair
+	 * @param {Object} DKIMSignature
+	 *			DKIMSignature of the message
+	 * @returns {void}
+	 */	
+	checkHeadersSigned: function Policy_checkHeadersSigned(msgHeaders, DKIMSignature) {
+		
+		const POLICY_DKIM_UNSIGNED_HEADERS_WARNING_MODE = {
+			RELAXED : 10,
+			RECOMMENDED : 20,
+			STRICT : 30
+		}
+		
+		// The list of recommended headers to sign is mostly based on
+		// https://www.rfc-editor.org/rfc/rfc6376.html#section-5.4.
+		const SIGNEDHEADERS = {
+			REQUIRED : ["From", "Subject"],
+			RECOMMENDED : ["Date", "To", "Cc", "Resent-Date", "Resent-From", "Resent-To", "Resent-Cc", "In-Reply-To", "References", "List-Id", "List-Help", "List-Unsubscribe", "List-Subscribe", "List-Post", "List-Owner", "List-Archive"],
+			DESIRED : ["Message-ID", "Sender", "MIME-Version", "Content-Transfer-Encoding", "Content-Disposition", "Content-ID", "Content-Description", "Reply-To"]
```

Comment:
Is there a reason `Reply-To` is in the list, even if it will be pushed later down below to the `DESIRED` or `RECOMMENDED` array?

Reply:
No, I forgot to remove the header after implementing the logic to handle the reply-to header.